client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -714,6 +726,16 @@ Params withRefreshPolicy(RefreshPolicy refreshPolicy) {
             return this;
         }
 
+        Params withRequestsPerSecond(float requestsPerSecond) {
+            // the default in AbstractBulkByScrollRequest is Float.POSITIVE_INFINITY,
+            // but we don't want to add that to the URL parameters, instead we use -1
+            if (Float.isFinite(requestsPerSecond)) {
```

Comment:
I don't think we need to add it to the url at all if it is POSITIVE_INFINITY.

Reply:
We seem to need to add _something_,  because I got errors from RestRethrottleAction#prepareRequest if not setting the parameter at all. AbstractBaseReindexRestHandler.parseRequestsPerSecond seems to be happy with -1 though.

Reply:
I think it only needs to add `-1` to the rethrottle API. Reindex doesn't need it if it is the default. But I suppose it doesn't hurt to pass it. I withdraw my comment.